raiden_contracts/tests/test_channel_settle.py

```diff
@@ -389,6 +390,14 @@ def test_settlement_with_unauthorized_token_transfer(
         vals_B,
     )
 
+    # Assign additional tokens to A
+    assign_tokens(A, externally_transferred_amount)
```

Comment:
Because the test was completely wrong. `A` did not have tokens to transfer and the `transfer` function used does not revert, it just returns `false`. So `A` was not transferring anything to the `token_network` and the last balance checks were constructed wrongly.
I am adding some more checks for the balances in the test, so things are clear.

Reply:
Added the additional checks in https://github.com/raiden-network/raiden-contracts/pull/270/commits/dab0a9a0eca0828e0f4b53b1fb8035dab8e5c7fa